### PR TITLE
Adding a lot of missing method and argument descriptions

### PR DESCRIPTION
**Pull request by toji**
Also did some normalization of some verbiage and various minor formatting things (indentation of method definitions, number of line breaks, etc.)

***
<a href="https://pr-preview.s3.amazonaws.com/toji/gpuweb/pull/1015.html" title="Last updated on Aug 20, 2020, 12:42 AM UTC (7b4ed7e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/1015/c61c344...toji:7b4ed7e.html" title="Last updated on Aug 20, 2020, 12:42 AM UTC (7b4ed7e)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -1457,60 +1456,61 @@ interface GPUBufferUsage {
...
 
-        Note: it is valid to set {{GPUBufferDescriptor/mappedAtCreation}} to true without {{GPUBufferUsage/MAP_READ}}
-        or {{GPUBufferUsage/MAP_WRITE}} in {{GPUBufferDescriptor/usage}}. This can be used to set the buffer's
-        initial data.
+            Note: it is valid to set {{GPUBufferDescriptor/mappedAtCreation}} to true without {{GPUBufferUsage/MAP_READ}}
+            or {{GPUBufferUsage/MAP_WRITE}} in {{GPUBufferDescriptor/usage}}. This can be used to set the buffer's
+            initial data.
 
-    </div>
+        </div>
```
Comment (kvark) [inline spec/index.bs:1513]: is this the desired indentation? looks like 8 spaces here

File: spec/index.bs [hunk 2]
```diff
@@ -3445,18 +3453,17 @@ Issue: link to the semantics of SV_SampleIndex and SV_Coverage in WGSL spec.
...
             **Called on:** {{GPUDevice}} |this|.
 
             **Arguments:**
             <pre class=argumentdef for="GPUDevice/createRenderPipeline(descriptor)">
-                |descriptor|:
+                |descriptor|: Description of the {{GPURenderPipeline}} to create.
             </pre>
 
-            **Returns:** {{GPUBuffer}}
+            **Returns:** {{GPURenderPipeline}}
```
Comment (kvark) [inline spec/index.bs:3466]: :laughing:

File: spec/index.bs [hunk 3]
```diff
@@ -5993,10 +5993,10 @@ GPUQueue includes GPUObjectBase;
 
             **Arguments:**
             <pre class=argumentdef for="GPUQueue/writeTexture(destination, data, dataLayout, size)">
-                |destination|:
-                |data|:
-                |dataLayout|:
-                |size|:
+                |destination|: The texture [=subresource=] and origin to write to.
```
Comment (kvark) [inline spec/index.bs:5996]: nit: may be better to say `[=texture subresource=]`  which is also defined

File: spec/index.bs [hunk 4]
```diff
@@ -6047,9 +6047,9 @@ GPUQueue includes GPUObjectBase;
 
             **Arguments:**
             <pre class=argumentdef for="GPUQueue/copyImageBitmapToTexture(source, destination, copySize)">
-                source:
-                |destination|:
-                |copySize|:
+                |source|: {{ImageBitmap}} and origin to copy to |destination|.
+                |destination|: The texture [=subresource=] and origin to write to.
```
Comment (kvark) [inline spec/index.bs:6051]: same here

File: spec/index.bs [hunk 5]
```diff
@@ -6136,14 +6137,15 @@ Completion of a fence is signaled from the {{GPUQueue}} that created it.
 <dl dfn-type=method dfn-for=GPUQueue>
     : <dfn>signal(fence, signalValue)</dfn>
     ::
+        Signals the given fence, increasing it's completed value to `signalValue`.
```
Comment (kainino0x) [inline spec/index.bs:6140]: nit: Not sure what the best way will be to write the prose for all of the commands, but probably something like "Enqueues a command which signals..."

File: spec/index.bs [hunk 6]
```diff
@@ -6157,6 +6159,7 @@ The completion of the fence and the value it completes with can be observed from
 <dl dfn-type=method dfn-for=GPUFence>
     : <dfn>getCompletedValue()</dfn>
     ::
+        Returns the fence's current completion value.
```
Comment (kainino0x) [inline spec/index.bs:6162]: The client side state here isn't defined yet, but we can be more specific here anyway, e.g: "Returns the largest signalled value that has propagated to the [=content timeline=]."

File: spec/index.bs (hunk 5 above)
Comment (toji) [reply spec/index.bs:6140]: Fine by me, though that implies that we need to do a pass of the whole spec to reflect that language. Follow up PR?

File: spec/index.bs (hunk 1 above)
Comment (toji) [reply spec/index.bs:1513]: Yeah, this is consistent with the rest of the `<dl>` blocks and prevents a (minor) style issue.

File: spec/index.bs (hunk 2 above)
Comment (toji) [reply spec/index.bs:3480]: 😅

File: spec/index.bs (hunk 5 above)
Comment (kainino0x) [reply spec/index.bs:6230]: Definitely not now. We should eventually figure out how we should write these, with some things in mind:
- Somewhere it should be clear that the functions on the encoders just add some data onto the end of a list, and in `submit` we actual execute the commands in that list.
- We need to put the "how to execute the command" algorithm somewhere. Perhaps it's just inside the function definitions, in the form of a closure? like "Enqueue an item onto the encoder which, when executed, performs the following tasks"

Not urgent though.

**Pull request merged**